methods.rb

```diff
@@ -0,0 +1,123 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+    i = 0
+    my_each do |item|
+      yield item, i
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+    arry = []
+    for i in self
+      arry << i if yield i
+    end
+    arry
+  end
+
+  def my_all?(arg = nil)
+    output = false
+    filtered_array = if !arg # no arguments
+                       block_given? ? my_select { |el| yield(el) } : my_select { |el| el }
+                     elsif arg.is_a?(Regexp)
+                       my_select { |el| arg.match(el) }
+                     elsif arg.is_a?(Class)
+                       my_select { |el| el.class <= arg }
+                     else
+                       my_select { |el| el == arg }
+                     end
+    output = true if filtered_array == to_a
+    output
+  end
+
+  def my_any?(arg = nil)
+    output = false
+    filtered_array = if !arg # no arguments
+                       block_given? ? my_select { |el| yield(el) } : my_select { |el| el }
+                     elsif arg.is_a?(Regexp)
+                       my_select { |el| arg.match(el) }
+                     elsif arg.is_a?(Class)
+                       my_select { |el| el.class <= arg }
+                     else
+                       my_select { |el| el == arg }
+                     end
+    output = true unless filtered_array.to_a.empty?
+    output
+  end
+
+  def my_none?(arg = nil)
+    output = false
+    filtered_array = if !arg
+                       block_given? ? my_select { |el| yield(el) } : my_select { |el| el }
+                     elsif arg.is_a?(Regexp)
+                       my_select { |el| arg.match(el) }
+                     elsif arg.is_a?(Class)
+                       my_select { |el| el.class <= arg }
+                     else
+                       my_select { |el| el == arg }
+                     end
+    output = true if filtered_array.to_a.empty?
+    output
+  end
+
+  def my_count (argument=nil)
+   counter = 0
+   if block_given?
+    to.a.my_each {|item| counter += 1 if yield(item)}
+   elsif !argument.nil?
+    to_a.my_each{|item| counter += 1 if argument == item}
+  else
+    counter = to_a.length
+  end
+  counter
+  end  
+
+
+  def my_map(proc_block = nil)
+    return to_enum(:my_map) unless block_given?
+
+    new_arr = []
+
+    if proc_block.class == Proc and block_given?
+      my_each { |el| new_arr.push(proc_block.call(el)) }
+    else
+      my_each { |el| new_arr.push(yield(el)) }
+    end
+
+    new_arr
+  end
+
+  def my_inject(init = nil, arg = nil)
```

Comment:
- [ ] enumerables #my_inject when a symbol is specified without an initail value combines each element of the collection by applying the symbol as a named method
![inject](https://user-images.githubusercontent.com/33067256/98528093-e2287480-228c-11eb-83b9-338d4350c8ce.png)

- [ ]  enumerables #my_inject when a symbol is specified without an initial value, it combines each element of the collection by applying the symbol as a named method: range

![range](https://user-images.githubusercontent.com/33067256/98528400-49debf80-228d-11eb-8404-5c1a3c4ac6e7.png)